src/main/java/org/prebid/server/bidder/amx/AmxBidder.java

```diff
@@ -54,8 +55,17 @@
 
     public AmxBidder(String endpointUrl, JacksonMapper mapper) {
         this.mapper = Objects.requireNonNull(mapper);
-        this.endpointUrl = new URIBuilder()
-                .setPath(HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl)))
+        this.endpointUrl = resolveEndpointUrl(endpointUrl);
+    }
+
+    private static String resolveEndpointUrl(String url) {
+        final URIBuilder uriBuilder;
+        try {
+            uriBuilder = new URIBuilder(HttpUtil.validateUrl(Objects.requireNonNull(url)));
```

Comment:
Seems we will already have validation for url while creating uri builder

Reply:
i'd say `new URI(string)` doesn't cover invalid url as `new URL(url)` does.
So, `org.prebid.server.bidder.amx.AmxBidderTest#creationShouldFailOnInvalidEndpointUrl` will fail.